kmir/src/kmir/kmir.py

```diff
@@ -1,231 +1,296 @@
+    backend_cmd: list[str]  # `kore-rpc` for legacy backend; `kore-booster-rpc ` for booster backend with dylib supplied
+    bug_report: Optional[BugReport] = None
+
+    # use_booster should be removed in future
+    def __init__(self, haskell_dir: Path, use_booster: bool, br: Optional[BugReport]):
+        mir_prove = KProve(haskell_dir)  # TODO: should pass in main-file, bug_report, save_directory etc.
+
+        if use_booster:
+            ext: str
+            match sys.platform:
+                case 'linux':
+                    ext = 'so'
+                case 'darwin':
+                    ext = 'dylib'
+                case _:
+                    raise ValueError('Unsupported platform: {sys.platform}')
+
+            kompiled_dir = self.haskell_dir / '..'
+            dylib = kompiled_dir / 'llvmc' / f'interpreter.{ext}'
+            check_file_path(dylib)
+
+            # TODO: Should use llvmc dir
+            llvm_dir = kompiled_dir / 'llvm'
+            llvm_definition = llvm_dir / 'llvm' / 'definition.kore'
+            llvm_dt = llvm_dir / 'dt'
+
+            cmd = ['kore-rpc-booster', '--llvm-backend-library', str(dylib)]
+            # _gather_booster_report(llvm_dir, llvm_definition, llvm_dt, br) if br else None
+
+            if br:
+                br.add_file(llvm_definition, Path('llvm_definition/definition.kore'))
+                br.add_file(llvm_dt, Path('llvm_definition/dt'))
+                llvm_version = run_process('llvm-backend-version', pipe_stderr=True, logger=_LOGGER).stdout.strip()
+                br.add_file_contents(llvm_version, Path('llvm_version.txt'))
+        else:
+            cmd = ['kore-rpc']
 
-    @staticmethod
-    def terminal_rules() -> list[str]:
-        terminal_rules = ['MIR.halt', 'MIR.panic', 'MIR.iPanic']
+        object.__setattr__(self, 'haskell_dir', haskell_dir)
+        object.__setattr__(self, 'mir_prove', mir_prove)
+        object.__setattr__(self, 'backend_cmd', cmd)
+        object.__setattr__(self, 'bug_report', br)
 
-        # TODO: break every step and add to terminal rules. Semantics does not support this currently
-        return terminal_rules
+    # start the kore server, `backend_cmd` decide which server to start.
+    def set_kore_server(
+        self,
+        *,
+        port: int | None = None,  # it is a positional argument for KoreClient and cannot be None.
+        smt_timeout: int | None = None,
+        smt_retry_limit: int | None = None,
+    ) -> KoreServer:
+        # Old way of handling KCFGExplore, to be removed. Is this still true?
+        kore_server_args: KoreServerArgs = {
+            'kompiled_dir': self.haskell_dir,
+            'module_name': self.mir_prove.main_module,
+            'port': port,
+            'command': self.backend_cmd,
+            'smt_timeout': smt_timeout,
+            'smt_retry_limit': smt_retry_limit,
+            'bug_report': self.bug_report,
+        }
+
+        return KoreServer(kore_server_args)
```

Comment:
@yanliu18 This change is from the update to pyk

Reply:
Thanks Daniel.